scripts/util_functions.sh

```diff
@@ -209,6 +209,11 @@ mount_partitions() {
     SYSTEM_ROOT=true
     setup_mntpoint /system_root
     mount --move /system /system_root
+    if [ $? != 0 ]; then
+      umount /system
+      umount -l /system 2>/dev/null
+      mount -o ro -t auto /dev/block/bootdevice/by-name/system$SLOT /system_root
```

Comment:
This will fail on devices with dynamic partitions (it failed on my Redmi K30 5G). Should use `mount_ro_ensure "system$SLOT app$SLOT" /system` just like how it mounts `/system` in the first place.